threedi_scenario_downloader/downloader.py

```diff
@@ -102,15 +102,21 @@ def get_scenario_instance(scenario_uuid):
     return scenario_instance
 
 
-def get_scenario_instance_results(scenario_uuid):
+def get_scenario_instance_results(scenario_uuid, subendpoint=None):
     """check if scenario exist"""
     get_scenario_instance(scenario_uuid)
 
+    if subendpoint:
+        url = "{}scenarios/{}/results/{}".format(LIZARD_URL, scenario_uuid, subendpoint)
+    else:
+        url = "{}scenarios/{}/results".format(LIZARD_URL, scenario_uuid)
+
     """return request response with all raster results"""
-    r = requests.get(
-        url="{}scenarios/{}/results".format(LIZARD_URL, scenario_uuid),
-        auth=("__key__", get_api_key()),
-    )
+    r = requests.get(url=url, auth=("__key__", get_api_key()))
+    if not r.json()["results"]:
+        logging.debug("The result data you request is non-existent")
+        raise ValueError("The result data you request is non-existent")
```

Comment:
Is dit de enige reden waarom results leeg kan zijn? Anders zie je straks in de logging "de boel bestaat niet" terwijl het een andere error is. Geen idee, ik kijk "los" naar de code.

Reply:
Goed punt. Als er iets mis is krijg je een raise status. Als het leeg is, zijn of bestaan de resultaten niet, of heb je geen rechten/autoriteit. De melding iets uitgebreidt.